### PR TITLE
Foundry and Functional testing

### PR DESCRIPTION
**Pull request by Cenadros**
- Implemented zenstruck/browser for e2e testing
- Implemented zenstruck/foundry for testing
- Updated .env.test with file database in sqlite
- Updated phpunit.xml.dist with two different testsuites (Unit and E2E)
- Created ContentStory to emulate insertion migration for testing, and loaded in test/bootstrap.php

### CODE REVIEW

File: config/bundles.php [hunk 1]
```diff
@@ -35,11 +35,12 @@
...
     Symfony\Bundle\DebugBundle\DebugBundle::class => ['dev' => true],
-    Symfony\Bundle\MakerBundle\MakerBundle::class => ['dev' => true],
+    Symfony\Bundle\MakerBundle\MakerBundle::class => ['dev' => true, 'test' => true],
     Symfony\Bundle\WebProfilerBundle\WebProfilerBundle::class => ['dev' => true],
     Doctrine\Bundle\FixturesBundle\DoctrineFixturesBundle::class => ['test' => true],
     Fidry\AliceDataFixtures\Bridge\Symfony\FidryAliceDataFixturesBundle::class => ['test' => true],
     Nelmio\Alice\Bridge\Symfony\NelmioAliceBundle::class => ['test' => true],
     SymfonyCasts\Bundle\ResetPassword\SymfonyCastsResetPasswordBundle::class => ['all' => true],
     Runroom\UserBundle\RunroomUserBundle::class => ['all' => true],
+    Zenstruck\Foundry\ZenstruckFoundryBundle::class => ['dev' => true],
```
Comment (aerrasti) [inline config/bundles.php:45]: Maybe is not working because Zenstruck is not enabled in test ?

```suggestion
    Zenstruck\Foundry\ZenstruckFoundryBundle::class => ['dev' => true, 'test' => true],
```

File: src/Story/ContentStory.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            'createdAt' => new \DateTime(),
+        ]);
+
+        // Create meta information
+        foreach ($this->getMetaInformationData() as $meta) {
+            $translations = $meta['translations'];
+            $meta['meta']['translations'] = MetaInformationTranslationFactory::new(function () use (&$translations) {
+                return array_pop($translations);
+            })->many(\count($translations));
+            MetaInformationFactory::new($meta['meta'])->create();
```
Comment (aerrasti) [inline src/Story/ContentStory.php:37]: You can use ::createOne instead of ::new and ->create()

File: composer.json [hunk 3]
```diff
@@ -112,6 +114,7 @@
         "lint-yaml": "bin/console lint:yaml config src translations --parse-tags",
         "php-cs-fixer": "php-cs-fixer fix --ansi --verbose",
         "phpstan": "phpstan analyse --no-progress",
-        "psalm": "psalm --stats"
+        "psalm": "psalm --stats",
+        "testDebug": "PHP_IDE_CONFIG=\"serverName=localhost\" exec php -dxdebug.mode=debug -dxdebug.start_with_request=1 -dxdebug.client_host=host.docker.internal -dxdebug.client_port=9003 -dremote_connect_back=0 vendor/bin/phpunit"
```
Comment (jordisala1991) [inline composer.json:118]: can you revert this change?

File: config/packages/doctrine.yaml [hunk 4]
```diff
@@ -32,3 +32,21 @@ when@prod:
                     adapter: cache.app
                 doctrine.system_cache_pool:
                     adapter: cache.system
+
+when@test:
```
Comment (jordisala1991) [inline config/packages/doctrine.yaml:36]: why do we need this change?

File: config/packages/framework.yaml [hunk 5]
```diff
@@ -15,3 +15,5 @@ when@test:
         test: true
         session:
             storage_factory_id: session.storage.factory.mock_file
+        assets:
+            json_manifest_path: ~
```
Comment (jordisala1991) [inline config/packages/framework.yaml:19]: Probably we should only do this on dev and test

File: config/packages/webpack_encore.yaml [hunk 6]
```diff
@@ -10,3 +10,7 @@ when@prod:
 when@test:
     webpack_encore:
         strict_mode: false
+
+when@dev:
```
Comment (jordisala1991) [inline config/packages/webpack_encore.yaml:14]: can you put this before when@test?

File: phpunit.xml.dist [hunk 7]
```diff
@@ -14,15 +14,22 @@
     xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
 >
     <testsuites>
-        <testsuite name="Archetype Symfony Test Suite">
-            <directory>tests</directory>
+        <testsuite name="Unit">
+            <directory>tests/Unit</directory>
+        </testsuite>
+        <testsuite name="E2E">
```
Comment (jordisala1991) [inline phpunit.xml.dist:20]: I would prefer naming them : Functional, can you replace everything related to E2E?

**Review by jordisala1991 — COMMENTED** [review]
Nice work, some minor comments

File: src/Story/ContentStory.php (hunk 2 above)
Comment (Cenadros) [reply src/Story/ContentStory.php]: Thanks! I changed it.

File: composer.json (hunk 3 above)
Comment (Cenadros) [reply composer.json]: Done!

File: config/packages/doctrine.yaml (hunk 4 above)
Comment (Cenadros) [reply config/packages/doctrine.yaml:36]: I did improve this part to only include the `schema_filter` part. We need this change because of how the `ResetDatabase` trait works. When it drops the database in order to create it again, it fails because of a table being specified in the `schema_filter` (it already exists in the database).

File: config/packages/webpack_encore.yaml (hunk 6 above)
Comment (Cenadros) [reply config/packages/webpack_encore.yaml]: Done!

File: phpunit.xml.dist (hunk 7 above)
Comment (Cenadros) [reply phpunit.xml.dist]: Done!

**Review by jordisala1991 — APPROVED** [review]
